### PR TITLE
[nanoflann] Add new port

### PR DESCRIPTION
**Pull request by AlvinZhangH**
Related issue: #8784. There are no features of this port need to test.

### CODE REVIEW

File: ports/nanoflann/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO jlblancoc/nanoflann
+    REF e8792e464ab05267216acde8b4ddf301714176a2 #1.3.1
+    SHA512 78a04d39b418b6c6582e6d4180958bb0b492547a9662026da07a8b75d7186140bc4d6b50b6eece32db0196607cfcc901aaf4b458e9ab8a9a115b569acc2bae40
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+)
```
Comment (JackBoosY) [inline ports/nanoflann/portfile.cmake]: Add `PREFER_NINJA` to use ninja.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/nanoflann/portfile.cmake (hunk 1 above)
Comment (AlvinZhangH) [reply ports/nanoflann/portfile.cmake]: @JackBoosY Already added.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**